### ISSUE TITLE
Postfix Connection timed out

### ISSUE THREAD

**Issue by shaneonabike** [mailcow/mailcow-dockerized#1010]
I had been struggling with getting this setup behind a router, but I _think_ at this point it's working.

For some bizarre reason the emails are not being sent out. I opened all the ports on the router and they aren't blocked on the ufw configuration (using Ubuntu 16.04). Any clues why I might be getting timeouts?

> postfix-mailcow_1    | Feb  7 11:17:56 mail postfix/qmgr[360]: ED341740998: from=<test@blah.media>, size=1103, nrcpt=1 (queue active)
> postfix-mailcow_1    | Feb  7 11:18:26 mail postfix/smtp[396]: connect to mail.doolittle.com[45.56.102.194]:25: Connection timed out
> postfix-mailcow_1    | Feb  7 11:18:26 mail postfix/smtp[396]: EFE2274079B: to=<shane@doolittle.com>, relay=none, delay=1145, delays=1115/0.05/30/0, dsn=4.4.1, status=deferred (connect to mail.doolittle.com[45.56.102.194]:25: Connection timed out)
> postfix-mailcow_1    | Feb  7 11:18:26 mail postfix/smtp[395]: connect to mail.blah.media[70.30.65.99]:25: Connection timed out
> postfix-mailcow_1    | Feb  7 11:18:26 mail postfix/smtp[395]: ED341740998: to=<test@lah.media>, relay=none, delay=7691, delays=7660/0.05/30/0, dsn=4.4.1, status=deferred (connect to mail.blah.media[70.30.65.99]:25: Connection timed out)

I am able to telnet directly to the port 25 properly remotely and within the server as well...

**Comment by andryyy** [comment]
Ufw? Are you sure Dockers netfilter rules are not removed? Could also be a Mac spoofing filter.

**Comment by shaneonabike** [comment]
How would I verify that...? I checked the iptables and they are all in there.. but there isn't anything in ufw. What is the Mac spoofing filter?

Would it help to get an output of ufw and iptables?

**Comment by stevesbrain** [comment]
@shaneonabike Beyond that, are you on a residential line? If so, it's not just your ISP blocking access to port 25? That's a fairly common thing.

**Comment by shaneonabike** [comment]
I am on a residential line using no-ip with a registered domain IP @stevesbrain so technically it should work right? The email that I am trying to send to is on my own server. Also, worth noting that I cannot send emails to myself either so I feel that there is something not configured properly

I had to set docker to run listen on 127.0.0.1:8080 and 127.0.0.1:8443 for what it's worth. As well, I disabled IPV6 because it never could lookup the corresponding server properly, and I wanted to get it at least finding a connection which it does. The domain is indianroad.media

For what it's worth I can telnet to the server easily on all ports so that's a null issue.

### ufw
> To                         Action      From
> --                         ------      ----
> 80,443/tcp (Apache Full)   ALLOW IN    Anywhere                  
> 2048                       ALLOW IN    Anywhere                  
> 80                         ALLOW IN    Anywhere                  
> 443                        ALLOW IN    Anywhere                  
> 21/tcp                     ALLOW IN    Anywhere                  
> 80 on eth0                 ALLOW IN    Anywhere                  
> 443 on eth0                ALLOW IN    Anywhere                  
> 25                         ALLOW IN    Anywhere                  
> 110                        ALLOW IN    Anywhere                  
> 143                        ALLOW IN    Anywhere                  
> 465                        ALLOW IN    Anywhere                  
> 587                        ALLOW IN    Anywhere                  
> 993                        ALLOW IN    Anywhere                  
> 995                        ALLOW IN    Anywhere                  
> 80,443/tcp (Apache Full (v6)) ALLOW IN    Anywhere (v6)             
> 2048 (v6)                  ALLOW IN    Anywhere (v6)             
> 80 (v6)                    ALLOW IN    Anywhere (v6)             
> 443 (v6)                   ALLOW IN    Anywhere (v6)             
> 21/tcp (v6)                ALLOW IN    Anywhere (v6)             
> 80 (v6) on eth0            ALLOW IN    Anywhere (v6)             
> 443 (v6) on eth0           ALLOW IN    Anywhere (v6)             
> 25 (v6)                    ALLOW IN    Anywhere (v6)             
> 110 (v6)                   ALLOW IN    Anywhere (v6)             
> 143 (v6)                   ALLOW IN    Anywhere (v6)             
> 465 (v6)                   ALLOW IN    Anywhere (v6)             
> 587 (v6)                   ALLOW IN    Anywhere (v6)             
> 993 (v6)                   ALLOW IN    Anywhere (v6)             
> 995 (v6)                   ALLOW IN    Anywhere (v6)   

### iptables

Chain INPUT (policy ACCEPT)
target     prot opt source               destination         
ufw-before-logging-input  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-before-input  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-after-input  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-after-logging-input  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-reject-input  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-track-input  all  --  0.0.0.0/0            0.0.0.0/0           
Chain FORWARD (policy ACCEPT)
target     prot opt source               destination         
DOCKER-USER  all  --  0.0.0.0/0            0.0.0.0/0           
DOCKER-ISOLATION  all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
DOCKER     all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
DOCKER     all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
ufw-before-logging-forward  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-before-forward  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-after-forward  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-after-logging-forward  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-reject-forward  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-track-forward  all  --  0.0.0.0/0            0.0.0.0/0           
Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         
ufw-before-logging-output  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-before-output  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-after-output  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-after-logging-output  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-reject-output  all  --  0.0.0.0/0            0.0.0.0/0           
ufw-track-output  all  --  0.0.0.0/0            0.0.0.0/0           
Chain DOCKER (2 references)
target     prot opt source               destination         
ACCEPT     tcp  --  0.0.0.0/0            172.22.1.5           tcp dpt:587
ACCEPT     tcp  --  0.0.0.0/0            172.22.1.5           tcp dpt:465
ACCEPT     tcp  --  0.0.0.0/0            172.22.1.5           tcp dpt:25
ACCEPT     tcp  --  0.0.0.0/0            172.22.1.6           tcp dpt:12345
ACCEPT     tcp  --  0.0.0.0/0            172.22.1.6           tcp dpt:4190
ACCEPT     tcp  --  0.0.0.0/0            172.22.1.6           tcp dpt:995
ACCEPT     tcp  --  0.0.0.0/0            172.22.1.6           tcp dpt:993
ACCEPT     tcp  --  0.0.0.0/0            172.22.1.6           tcp dpt:143
ACCEPT     tcp  --  0.0.0.0/0            172.22.1.6           tcp dpt:110
ACCEPT     tcp  --  0.0.0.0/0            172.22.1.11          tcp dpt:8443
ACCEPT     tcp  --  0.0.0.0/0            172.22.1.11          tcp dpt:8080
Chain DOCKER-ISOLATION (1 references)
target     prot opt source               destination         
DROP       all  --  0.0.0.0/0            0.0.0.0/0           
DROP       all  --  0.0.0.0/0            0.0.0.0/0           
RETURN     all  --  0.0.0.0/0            0.0.0.0/0           
Chain DOCKER-USER (1 references)
target     prot opt source               destination         
RETURN     all  --  0.0.0.0/0            0.0.0.0/0           
Chain ufw-after-forward (1 references)
target     prot opt source               destination         
Chain ufw-after-input (1 references)
target     prot opt source               destination         
ufw-skip-to-policy-input  udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:137
ufw-skip-to-policy-input  udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:138
ufw-skip-to-policy-input  tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:139
ufw-skip-to-policy-input  tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:445
ufw-skip-to-policy-input  udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:67
ufw-skip-to-policy-input  udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:68
ufw-skip-to-policy-input  all  --  0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type BROADCAST
Chain ufw-after-logging-forward (1 references)
target     prot opt source               destination         
LOG        all  --  0.0.0.0/0            0.0.0.0/0            limit: avg 3/min burst 10 LOG flags 0 level 4 prefix "[UFW BLOCK] "
Chain ufw-after-logging-input (1 references)
target     prot opt source               destination         
LOG        all  --  0.0.0.0/0            0.0.0.0/0            limit: avg 3/min burst 10 LOG flags 0 level 4 prefix "[UFW BLOCK] "
Chain ufw-after-logging-output (1 references)
target     prot opt source               destination         
Chain ufw-after-output (1 references)
target     prot opt source               destination         
Chain ufw-before-forward (1 references)
target     prot opt source               destination         
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0            icmptype 3
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0            icmptype 4
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0            icmptype 11
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0            icmptype 12
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0            icmptype 8
ufw-user-forward  all  --  0.0.0.0/0            0.0.0.0/0           
Chain ufw-before-input (1 references)
target     prot opt source               destination         
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
ufw-logging-deny  all  --  0.0.0.0/0            0.0.0.0/0            ctstate INVALID
DROP       all  --  0.0.0.0/0            0.0.0.0/0            ctstate INVALID
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0            icmptype 3
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0            icmptype 4
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0            icmptype 11
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0            icmptype 12
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0            icmptype 8
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp spt:67 dpt:68
ufw-not-local  all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     udp  --  0.0.0.0/0            224.0.0.251          udp dpt:5353
ACCEPT     udp  --  0.0.0.0/0            239.255.255.250      udp dpt:1900
ufw-user-input  all  --  0.0.0.0/0            0.0.0.0/0           
Chain ufw-before-logging-forward (1 references)
target     prot opt source               destination         
Chain ufw-before-logging-input (1 references)
target     prot opt source               destination         
Chain ufw-before-logging-output (1 references)
target     prot opt source               destination         
Chain ufw-before-output (1 references)
target     prot opt source               destination         
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
ufw-user-output  all  --  0.0.0.0/0            0.0.0.0/0           
Chain ufw-logging-allow (0 references)
target     prot opt source               destination         
LOG        all  --  0.0.0.0/0            0.0.0.0/0            limit: avg 3/min burst 10 LOG flags 0 level 4 prefix "[UFW ALLOW] "
Chain ufw-logging-deny (2 references)
target     prot opt source               destination         
RETURN     all  --  0.0.0.0/0            0.0.0.0/0            ctstate INVALID limit: avg 3/min burst 10
LOG        all  --  0.0.0.0/0            0.0.0.0/0            limit: avg 3/min burst 10 LOG flags 0 level 4 prefix "[UFW BLOCK] "
Chain ufw-not-local (1 references)
target     prot opt source               destination         
RETURN     all  --  0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL
RETURN     all  --  0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type MULTICAST
RETURN     all  --  0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type BROADCAST
ufw-logging-deny  all  --  0.0.0.0/0            0.0.0.0/0            limit: avg 3/min burst 10
DROP       all  --  0.0.0.0/0            0.0.0.0/0           
Chain ufw-reject-forward (1 references)
target     prot opt source               destination         
Chain ufw-reject-input (1 references)
target     prot opt source               destination         
Chain ufw-reject-output (1 references)
target     prot opt source               destination         
Chain ufw-skip-to-policy-forward (0 references)
target     prot opt source               destination         
DROP       all  --  0.0.0.0/0            0.0.0.0/0           
Chain ufw-skip-to-policy-input (7 references)
target     prot opt source               destination         
DROP       all  --  0.0.0.0/0            0.0.0.0/0           
Chain ufw-skip-to-policy-output (0 references)
target     prot opt source               destination         
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
Chain ufw-track-forward (1 references)
target     prot opt source               destination         
Chain ufw-track-input (1 references)
target     prot opt source               destination         
Chain ufw-track-output (1 references)
target     prot opt source               destination         
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            ctstate NEW
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            ctstate NEW
Chain ufw-user-forward (1 references)
target     prot opt source               destination         
Chain ufw-user-input (1 references)
target     prot opt source               destination         
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            multiport dports 80,443 /* 'dapp_Apache%20Full' */
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:2048
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:2048
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:80
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:80
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:443
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:443
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:21
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:80
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:80
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:443
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:443
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:25
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:25
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:110
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:110
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:143
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:143
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:465
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:465
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:587
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:587
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:993
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:993
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:995
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:995

Chain ufw-user-limit (0 references)
target     prot opt source               destination         
LOG        all  --  0.0.0.0/0            0.0.0.0/0            limit: avg 3/min burst 5 LOG flags 0 level 4 prefix "[UFW LIMIT BLOCK] "
REJECT     all  --  0.0.0.0/0            0.0.0.0/0            reject-with icmp-port-unreachable

Chain ufw-user-limit-accept (0 references)
target     prot opt source               destination         
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           

Chain ufw-user-logging-forward (0 references)
target     prot opt source               destination         

Chain ufw-user-logging-input (0 references)
target     prot opt source               destination         

Chain ufw-user-logging-output (0 references)
target     prot opt source               destination         

Chain ufw-user-output (1 references)
target     prot opt source               destination 

### netstat
> Active Internet connections (only servers)
> Proto Recv-Q Send-Q Local Address           Foreign Address         State       User       Inode       PID/Program name
> tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      0          1277612     17927/cupsd     
> tcp        0      0 127.0.0.1:19991         0.0.0.0:*               LISTEN      0          894859      20566/docker-proxy
> tcp        0      0 127.0.0.1:8443          0.0.0.0:*               LISTEN      0          905276      21029/docker-proxy
> tcp        0      0 0.0.0.0:2048            0.0.0.0:*               LISTEN      0          790785      1570/sshd       
> tcp        0      0 127.0.0.1:3306          0.0.0.0:*               LISTEN      111        25602       2008/mysqld     
> tcp        0      0 127.0.0.1:8080          0.0.0.0:*               LISTEN      0          904342      21041/docker-proxy
> tcp6       0      0 ::1:631                 :::*                    LISTEN      0          1277611     17927/cupsd     
> tcp6       0      0 :::25                   :::*                    LISTEN      0          894851      20554/docker-proxy
> tcp6       0      0 :::443                  :::*                    LISTEN      0          39675       5785/apache2    
> tcp6       0      0 :::4190                 :::*                    LISTEN      0          894876      20577/docker-proxy
> tcp6       0      0 :::2048                 :::*                    LISTEN      0          790787      1570/sshd       
> tcp6       0      0 :::993                  :::*                    LISTEN      0          899842      20614/docker-proxy
> tcp6       0      0 :::995                  :::*                    LISTEN      0          894890      20596/docker-proxy
> tcp6       0      0 :::587                  :::*                    LISTEN      0          899418      19693/docker-proxy
> tcp6       0      0 :::110                  :::*                    LISTEN      0          901722      20638/docker-proxy
> tcp6       0      0 :::143                  :::*                    LISTEN      0          901708      20626/docker-proxy
> tcp6       0      0 :::80                   :::*                    LISTEN      0          39671       5785/apache2    
> tcp6       0      0 :::465                  :::*                    LISTEN      0          903399      20542/docker-proxy
> udp        0      0 0.0.0.0:631             0.0.0.0:*                           0          1279095     17930/cups-browsed
> udp        0      0 0.0.0.0:53947           0.0.0.0:*                           114        15049       1236/avahi-daemon: 
> udp        0      0 0.0.0.0:5353            0.0.0.0:*                           114        15047       1236/avahi-daemon: 
> udp        0      0 0.0.0.0:68              0.0.0.0:*                           0          15223       1474/dhclient   
> udp6       0      0 :::5353                 :::*                                114        15048       1236/avahi-daemon: 
> udp6       0      0 :::47396                :::*                                114        15050       1236/avahi-daemon: 
`

**Comment by shaneonabike** [comment]
Actually on more research it seems that the ISP adds another layer of blocking darnit! Thanks @stevesbrain  for pointing that out and sorry for wasting peeps time

**Comment by stevesbrain** [comment]
@shaneonabike No worries at all - I've seen it happen a few times :) Best you could do if you want to actually run it on-premises would be to create a permanent VPN tunnel for it all to flow through!
